### PR TITLE
tests/subsys/settings/fcb_int: add storage preparation

### LINKED ISSUE #16463: tests/subsys/settings/fcb_init fails on second run
**Describe the bug**
build the tests/subsys/settings/fcb_init application, run and test_init fails

**To Reproduce**
1. mkdir build; cd build
2. cmake -DBOARD=twr_ke18f ..
3. make

**Expected behavior**
the setting_fcb test can PASS

**Impact**
settings fcb can not work

**Screenshots or console output**
```
***** Booting Zephyr OS 1.14.99 *****

Running test suite test_initialization

===================================================================

starting test - test_init



    Assertion failed at ./tests/subsys/settings/fcb_init/src/settings_test_fcb_init.c:61: test_init: (prev_int not equal to val32)

load value doesn't match to what was saved

FAIL - test_init

===================================================================

Test suite test_initialization failed.

===================================================================

PROJECT EXECUTION FAILED

```

**Environment (please complete the following information):**
 - OS: (e.g. Linux)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA : 59c8c9930d0b27e2a53e05cd5947cbe7f8bc800b

### PR DESCRIPTION
**Pull request by nvlsianpu**
The test assumes that storage is clear before it runs. This
might be not true as lot of devices are only partially erased by
'west flash'.

Patch introduce procedure for testing whether test runs the first
time and clear storage if so.

fixes #16463

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
LGTM

**Review by ioannisg — CHANGES_REQUESTED** [review]
Fix commit title typo :p

**Comment by nvlsianpu** [comment]
@ioannisg typo fixed

**Comment by ioannisg** [comment]
@hakehuang can you please try the patch and confirm the fix?

**Comment by ioannisg** [comment]
@nvlsianpu it is not clear to me (from the description in the commit) why the patches written are under CONFIG_ARM. This does not seem to be architecture-specific, so we need to elaborate a bit more on that.

**Comment by nvlsianpu** [comment]
@ioannisg 
Yes - solution is architecture specific - It will work on most of ARM SoC.
Yes, it is kind of temp-hack here to make stuff testable.

Will add description on that in commit message and code

**Comment by ioannisg** [comment]
> 
> 
> @ioannisg
> Yes - solution is architecture specific - It will work on most of ARM SoC.
> Yes, it is kind of temp-hack here to make stuff testable.
> 
> Will add description on that in commit message and code

I think we need something better here. CONFIG_ARM is really misleading; shows that something is required because of the Cortex-M architecture. We better have #ifdefs that are more close to reality, i.e. platform, SOC, or SOC-family specific (or Driver specific)

**Comment by nvlsianpu** [comment]
@ioannisg reworked

File: tests/subsys/settings/fcb_init/src/settings_test_fcb_init.c [hunk 1]
```diff
@@ -62,10 +77,52 @@ void test_init(void)
 		      "load value doesn't match to what was saved");
 }
 
+
+void test_prepare_storage(void)
+{
+#ifdef DT_FLASH_AREA_IMAGE_0_ID
+/* This procedure uses mark which is stored inside SoC embedded program
+ * flash. It will not work one devices on which read/write to it is
```
Comment (ioannisg) [inline tests/subsys/settings/fcb_init/src/settings_test_fcb_init.c]: on*

Comment (ioannisg) [reply tests/subsys/settings/fcb_init/src/settings_test_fcb_init.c]: read/write to them is not possible

**Review by ioannisg — COMMENTED** [review]
Some minor comments, otherwise looks ok to me

**Comment by nvlsianpu** [comment]
@ioannisg updated

**Review by ioannisg — APPROVED** [review]

**Pull request merged**